0.1.Copy data from Excel codes.R

```diff
@@ -1,79 +1,75 @@
-rm(list = ls(all.names = TRUE)) #will clear all objects includes hidden objects.
-gc() #free up memrory and report the memory usage.
+#rm(list = ls(all.names = TRUE)) #will clear all objects includes hidden objects.
+#gc() #free up memrory and report the memory usage.
 memory.limit(size = 80000000)
 
 library(xlsx)
 library(ggplot2)
 library(ggpmisc)
 library(readxl)
+library(dplyr)
+library(forcats)
+library(bayestestR)
 
+# set data locations in one place
+inputdata <- c(subjects="Subjects with Clamp Infor.xlsx"
+               ,dat0="Erit-2 Clamp Calculations Dr M with lean mass (003)nm updated with 206 215_2.xlsx");
 
 ## Extracting data from clamp sheet
-subject_infor <- read_excel("E:/2021-03-26 Bulk Fat RNAseq/Subjects with Clamp Infor.xlsx",
+subject_infor <- read_excel(inputdata['subjects'],
                             sheet = "Subjects.infor", col_types = c("numeric", "text"))
-col.location <- read_excel("E:/2021-03-26 Bulk Fat RNAseq/Subjects with Clamp Infor.xlsx",
+col.location <- read_excel(inputdata['subjects'],
                             sheet = "col.location")
 
-for ( i in 1:nrow(subject_infor)) {
-  i=i
-  x <- read_excel("E:/2021-03-26 Bulk Fat RNAseq/Erit-2 Clamp Calculations Dr M with lean mass (003)nm updated with 206 215_2.xlsx", 
-                  sheet = subject_infor$Sheet_Name[i], col_names = FALSE)
-  for (b in 1:nrow(col.location)) {
-    b =b
-    subject_infor[i, col.location$col[b]] <- x[col.location$nrow[b], col.location$ncol_number[b]]
-  }
-}
+dat0 <- NULL;
+for ( i in unique(subject_infor$Sheet_Name)) {
+  dat0 <- rbind(dat0
+               ,bind_cols(apply(col.location,1,function(thiscol){
+                   read_excel(inputdata['dat0'],sheet=i,col_names=F
+                                     ,range=thiscol['location']) %>%
+                   ifelse(nrow(.)==0,as_tibble(NA),.) %>%
+                   setNames(thiscol['col'])})))
+};
+dat0 <- cbind(subject_infor,dat0);
+
 
 ## organize the data extracted from clamp sheet
 # change date
-library("zoo")
-subject_infor$DATE <- as.Date(as.numeric(subject_infor$DATE), origin="1899-12-30")
+dat0$DATE <- as.Date(dat0$DATE);

```

Comment:
You don't need to do all that stuff you did before to change the `DATE` column to a date. In fact you probably don't need to convert date at all-- in this version it is imported as `"POSIXct" "POSIXt"` which is just a more precise version of the `Date` class.